src/Kurs/Skjema.elm

```diff
@@ -0,0 +1,463 @@
+module Kurs.Skjema exposing
+    ( Felt(..)
+    , FullførtDato(..)
+    , KursSkjema
+    , ValidertKursSkjema
+    , VarighetEnhet(..)
+    , encode
+    , feilmeldingKursholder
+    , feilmeldingKursnavn
+    , feilmeldingKursnavnHvisSynlig
+    , feilmeldingPeriode
+    , feilmeldingValgfrittFullførtÅr
+    , feilmeldingVarighet
+    , feilmeldingVarighetHvisSynlig
+    , fullførtDatoValidert
+    , fullførtMåned
+    , id
+    , initValidertSkjema
+    , innholdTekstFelt
+    , oppdaterFullførtMåned
+    , oppdaterTekstFelt
+    , oppdaterVarighetEnhet
+    , stringTilVarighetEnhet
+    , tilUvalidertSkjema
+    , tillatÅViseAlleFeilmeldinger
+    , tillatÅViseFeilmeldingFullførtÅr
+    , tillatÅViseFeilmeldingKursnavn
+    , tillatÅViseFeilmeldingVarighet
+    , valider
+    , validerFullført
+    , varighetEnhet
+    , varighetEnhetTilString
+    )
+
+import Dato exposing (Måned, År)
+import Json.Encode
+
+
+type KursSkjema
+    = UvalidertSkjema UvalidertSkjemaInfo
+
+
+type ValidertKursSkjema
+    = ValidertSkjema ValidertSkjemaInfo
+
+
+type FullførtDato
+    = IkkeOppgitt
+    | Oppgitt Måned År
+
+
+type VarighetEnhet
+    = TIME
+    | DAG
+    | UKE
+    | MND
```

Comment:
Du kan godt skrive disse med kun stor forbokstav, og skrive Måned i stedet for Mnd. Så lenge du konverterer dem til all caps strenger på vei til og fra serveren så kan de være hva som helst her.